core/src/main/java/com/opensymphony/xwork2/conversion/impl/DefaultConversionAnnotationProcessor.java

```diff
@@ -69,7 +70,7 @@ public void process(Map<String, Object> mapping, TypeConversion tc, String key)
                     mapping.put(key, tc.value());
                 }
                 //for properties of classes
-                else if (tc.rule() != ConversionRule.ELEMENT || tc.rule() == ConversionRule.KEY || tc.rule() == ConversionRule.COLLECTION) {
+                else if (tc.rule() != ConversionRule.ELEMENT && tc.rule() != ConversionRule.KEY && tc.rule() != ConversionRule.COLLECTION) {
```

Comment:
This is a fix for that issue which I discussed at [dev list](https://www.mail-archive.com/dev@struts.apache.org/msg43544.html). I suspected it should be wrong because this `else if` then does not require ` || tc.rule() == ConversionRule.KEY || tc.rule() == ConversionRule.COLLECTION` because it's always true when `tc.rule() != ConversionRule.ELEMENT`! Also, this `else if` was handling all `tc.rule() != ConversionRule.ELEMENT` and execution never reached it's next else if, `tc.rule() == ConversionRule.KEY`.

So i asked at dev list and your reply directed me that this code is copied from [DefaultConversionFileProcessor.java#78](https://github.com/apache/struts/blob/29b29a9c64aa4700b0f3c52f250ef27a9dc67516/core/src/main/java/com/opensymphony/xwork2/conversion/impl/DefaultConversionFileProcessor.java#L78). Please see there that a NOT is applied to all statement but the copier of code removed the NOT but did not applied on all but only on first. This is that second issue which is fixed here side by side WW-4906 :)